### ISSUE TITLE
New Channel need LND re-start to announce to graph.

### ISSUE THREAD

**Issue by BhaagBoseDK** [lightningnetwork/lnd#6882]
### Background

Since upgrading to 0.15.1-beta I and some others are facing a unique issue. Once a channel is opened (either initiated by me or opened to me by remote), it seems to get stuck in "confirming funding locked" and never gets announced to graph until I restart lnd. Here is the latest example, I will add more (historical or future) as and when I find them.

**Channel Initiated by Remote**
1. Channel Open by remote

```
2022-09-04 02:36:53.134 [INF] FNDG: Recv'd fundingRequest(amt=0.11040886 BTC, push=0 mSAT, delay=144, pendingId=cde52bb26051dffee584fc33e2b73b35b1881499ebba665bdf34ca80e128d197) from peer(03cf31364b9361f434b91f0e1943664f723e30c818b9a396bbbf5ef4284612c293)
2022-09-04 02:36:53.139 [INF] CHFD: Performing funding tx coin selection using 0 sat/kw as fee rate
2022-09-04 02:36:53.216 [INF] FNDG: Requiring 1 confirmations for pendingChan(cde52bb26051dffee584fc33e2b73b35b1881499ebba665bdf34ca80e128d197): amt=0.11040886 BTC, push_amt=0 mSAT, committype=tweakless, upfrontShutdown=
2022-09-04 02:36:53.217 [INF] FNDG: Sending fundingResp for pending_id(cde52bb26051dffee584fc33e2b73b35b1881499ebba665bdf34ca80e128d197)
2022-09-04 02:36:53.377 [INF] FNDG: completing pending_id(cde52bb26051dffee584fc33e2b73b35b1881499ebba665bdf34ca80e128d197) with ChannelPoint(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2)
2022-09-04 02:36:53.405 [INF] FNDG: sending FundingSigned for pending_id(cde52bb26051dffee584fc33e2b73b35b1881499ebba665bdf34ca80e128d197) over ChannelPoint(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2)
2022-09-04 02:36:53.407 [INF] CNCT: Creating new ChannelArbitrator for ChannelPoint(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2)
2022-09-04 02:36:53.418 [INF] CNCT: ChannelArbitrator(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2): starting state=StateDefault, trigger=chainTrigger, triggerHeight=752513
2022-09-04 02:36:53.424 [INF] NTFN: New spend subscription: spend_id=229, outpoint=86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2, script=0 1a2bb535ebead75e4f2942b4881a1899fab918d4e87603c04b9c0d2ab95acf6b, height_hint=752513
2022-09-04 02:36:53.424 [INF] NTFN: Dispatching historical spend rescan for outpoint=86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2, script=0 1a2bb535ebead75e4f2942b4881a1899fab918d4e87603c04b9c0d2ab95acf6b, start=752513, end=752513
2022-09-04 02:36:53.449 [INF] CNCT: Close observer for ChannelPoint(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2) active
2022-09-04 02:36:53.452 [INF] NTFN: New block epoch subscription
2022-09-04 02:36:53.457 [INF] NTFN: New confirmation subscription: conf_id=4, txid=86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35, num_confs=1 height_hint=752513
2022-09-04 02:36:53.459 [INF] FNDG: Waiting for funding tx (86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35) to reach 1 confirmations
2022-09-04 02:36:53.461 [INF] NTFN: Historical spend dispatch finished for request outpoint=86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2, script=0 1a2bb535ebead75e4f2942b4881a1899fab918d4e87603c04b9c0d2ab95acf6b (start=752513 end=752513) with details: <nil>
2022-09-04 02:36:53.469 [INF] CHBU: Updating on-disk multi SCB backup: num_old_chans=225, num_new_chans=226
2022-09-04 02:36:53.596 [WRN] CHBU: Replacing disk backup for ChannelPoint(d6a24e9fb889213cead5fdb52489d3952716d9ba4bff6a35fec595dbdac261b2:1) w/ newer version
2
```

2. Chanel Opened
```
2022-09-04 02:49:32.071 [INF] NTFN: Dispatching 1 confirmation notification for txid=86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35
2022-09-04 02:49:32.077 [INF] UTXN: Attempting to graduate height=752514: num_kids=0, num_babies=0
2022-09-04 02:49:32.078 [INF] FNDG: ChannelPoint(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2) is now active: ChannelID(351be2b1a22612daebc0fb692c71e2828c6f957ded05c55d1483cec370b9bd84)
2022-09-04 02:49:32.187 [INF] FNDG: Peer(03cf31364b9361f434b91f0e1943664f723e30c818b9a396bbbf5ef4284612c293) is online, sending FundingLocked for ChannelID(351be2b1a22612daebc0fb692c71e2828c6f957ded05c55d1483cec370b9bd84)
```

3. Channel unable to receive funding lock. Unable to find forwarding policies
```
2022-09-04 02:49:57.236 [INF] PEER: Peer(03cf31364b9361f434b91f0e1943664f723e30c818b9a396bbbf5ef4284612c293): loading ChannelPoint(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2)
2022-09-04 02:49:57.239 [WRN] PEER: Peer(03cf31364b9361f434b91f0e1943664f723e30c818b9a396bbbf5ef4284612c293): Unable to find our forwarding policy for channel 86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2, using default values
2022-09-04 02:49:57.239 [INF] HSWC: ChannelLink(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2): starting
2022-09-04 02:49:57.239 [INF] HSWC: Trimming open circuits for chan_id=752514:1289:2, start_htlc_id=0
2022-09-04 02:49:57.240 [INF] HSWC: Adding live link chan_id=351be2b1a22612daebc0fb692c71e2828c6f957ded05c55d1483cec370b9bd84, short_chan_id=752514:1289:2
2022-09-04 02:49:57.240 [INF] HSWC: ChannelLink(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2): HTLC manager started, bandwidth=0 mSAT
2022-09-04 02:49:57.240 [INF] HSWC: ChannelLink(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2): attempting to re-synchronize
2022-09-04 02:49:57.240 [INF] PEER: Peer(03cf31364b9361f434b91f0e1943664f723e30c818b9a396bbbf5ef4284612c293): Negotiated chan series queries
2022-09-04 02:49:57.240 [INF] DISC: Creating new GossipSyncer for peer=03cf31364b9361f434b91f0e1943664f723e30c818b9a396bbbf5ef4284612c293
2022-09-04 02:49:57.240 [INF] NTFN: New block epoch subscription
2022-09-04 02:49:57.240 [INF] CNCT: Attempting to update ContractSignals for ChannelPoint(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2)
2022-09-04 02:49:58.223 [INF] HSWC: ChannelLink(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2): resending FundingLocked message to peer
2022-09-04 02:49:58.224 [INF] HSWC: ChannelLink(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2): received re-establishment message from remote side
2022-09-04 02:49:58.312 [INF] PEER: Peer(03cf31364b9361f434b91f0e1943664f723e30c818b9a396bbbf5ef4284612c293): Already have ChannelPoint(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2), ignoring.
2022-09-04 02:49:58.312 [INF] PEER: Peer(03cf31364b9361f434b91f0e1943664f723e30c818b9a396bbbf5ef4284612c293): Processing retransmitted FundingLocked for ChannelPoint(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2)
2022-09-04 02:50:04.176 [ERR] NANN: Unable to retrieve chan status for Channel(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2): edge not found
2022-09-04 02:51:06.241 [ERR] PEER: Peer(03cf31364b9361f434b91f0e1943664f723e30c818b9a396bbbf5ef4284612c293): Unable to enable channel 86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2: edge not found
```

4. This goes on and on until I restart lnd

```
2022-09-04 02:53:31.163 [ERR] NANN: Unable to retrieve chan status for Channel(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2): edge not found
....
2022-09-04 07:47:55.181 [ERR] NANN: Unable to retrieve chan status for Channel(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2): edge not found

```

5. Restart lnd
```
2022-09-04 07:51:14.149 [ERR] FNDG: Unable to advance state(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2): failed to check if funding locked was received: funding manager shutting down
2022-09-04 07:51:14.837 [INF] LTND: Shutdown complete
2022-09-04 07:51:45.866 [INF] LTND: Version: 0.15.1-beta commit=v0.15.1-beta, build=production, logging=default, debuglevel=info

```
6. Channel Announced
```
2022-09-04 07:56:56.165 [INF] NTFN: New confirmation subscription: conf_id=3, txid=86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35, num_confs=6 height_hint=752514
2022-09-04 07:56:56.530 [INF] DISC: Received new remote channel announcement for 725154:2737:2
2022-09-04 07:56:56.545 [WRN] DISC: ignoring remote ChannelAnnouncement for own channel
2022-09-04 07:56:57.527 [INF] NTFN: Dispatching 6 confirmation notification for txid=86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35
2022-09-04 07:56:57.528 [INF] FNDG: Announcing ChannelPoint(86bdb970c3ce83145dc505ed7d956f8c82e2712c69fbc0ebda1226a2b1e21b35:2), short_chan_id=752514:1289:2
2022-09-04 07:56:57.529 [INF] DISC: Received new local channel announcement for 752514:1289:2
2022-09-04 07:56:57.829 [INF] DISC: Fully valid channel proof for short_chan_id=827397893148704770 constructed, adding to next ann batch
```

### Your environment

* version of `lnd`
```
umbrel@umbrel:~ $ lncli getinfo
{
    "version": "0.15.1-beta commit=v0.15.1-beta",
    "commit_hash": "fd1a95bf322cdd1c743a1554f8e470fbf9a5e564",
    "identity_pubkey": "03c5528c628681aa17ab9e117aa3ee6f06c750dfb17df758ecabcd68f1567ad8c1",
```

* which operating system (`uname -a` on *Nix)
`Linux umbrel 5.10.17-v8+ #1421 SMP PREEMPT Thu May 27 14:01:37 BST 2021 aarch64 GNU/Linux`

**Comment by BhaagBoseDK** [comment]
There is some coincidence that the peers in question were always CLN.

**Comment by TrezorHannes** [comment]
Can confirm that I had the same issue with a new peer-channel who opened to me.
Could not set the fee due to `edge not found` in the log. Updating fees threw an error
```
lncli updatechanpolicy --fee_rate_ppm 220 --chan_point 2e68b23f8c68253c9be4f8e857e49bb83aad5a9aa07bf1d3739ca62ef7b59c71:1 --base_fee_msat 1000 
 --time_lock_delta 144

{
        "failed_updates": [
                {
                        "outpoint": "2e68b23f8c68253c9be4f8e857e49bb83aad5a9aa07bf1d3739ca62ef7b59c71:1",
                        "reason": "UPDATE_FAILURE_UNKNOWN",
                        "update_error": "could not update policies"
                }
        ]
}
```
Restarting LND resolved the issue.

**Comment by indomitorum** [comment]
Had exactly the same problem yesterday when I opened a channel. v0.15.1-beta. Channel never got announced to graph until I restarted lnd. Quite a few operators have run into this issue lately after updating.

**Comment by blckbx** [comment]
Not sure if this is of help but just in case.
Also on 0.15.1, successfully opened a channel to a known CLN peer. And also fee updates work:
```
[INF] FNDG: Initiating fundingRequest(local_amt=0.1 BTC (subtract_fees=false), push_amt=0 mSAT, chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, peer=037e27d212432eaf499e4fb648d996944f3454c094dab36336bac573f82211a335, min_confs=1)
[INF] FNDG: Peer(037e27d212432eaf499e4fb648d996944f3454c094dab36336bac573f82211a335) is online, sending FundingLocked for ChannelID(2d5a2b9bdff0e8425717f97120c4b1780b73cb15762a65298f9cefa5940d5891)
[INF] PEER: Peer(037e27d212432eaf499e4fb648d996944f3454c094dab36336bac573f82211a335): New channel active ChannelPoint(90580d94a5ef9c8f29652a7615cb730b78b1c42071f9175742e8f0df9b2b5a2d:1) with peer
```
For reference: I'm experimenting with scid_alias option (zero conf = false) and by the time of channel opening, it was set to true: 
`protocol.option-scid-alias=true`. `listchannels` reported an assigned alias for this peer after opening.

**Comment by kpshukla** [comment]
Had exactly the same problem when I opened a channel with CLN peer. v0.15.1-beta. Channel never got announced to graph until I restarted lnd.

**Comment by BhaagBoseDK** [comment]
> Not sure if this is of help but just in case. Also on 0.15.1, successfully opened a channel to a known CLN peer. And also fee updates work:
> 

One difference could be that people reported this issue also run LNDg which runs a script to reconnect peer. If an inactive channel has a peer already connected, it is disconnected first and reconnected (hoping for the channel to be reactivated). Similar logic is used by bos reconnect.

Now the new channel is in inactive state until the peer confirms funding lock. If that is disconnected and reconnected while the funding lock is not received, does LND give up on resending the funding lock? The disconnect can happen naturally and not just because of LNDg script.

**Comment by derekross** [comment]
I personally did not have this issue, however two new LND users on the latest version experienced this issue last night. They were connecting to my LND 0.15.1-beta node and another LND 0.15.1-beta node. Neither of them were using LNDg or connecting to CoreLN as others mentioned. Restarting fixed the issue.

**Comment by Crypt-iQ** [comment]
I assume you have tried to reconnect? The problem seems to be that `NotifyWhenOnline` is hanging for some reason, which notifies certain functions when a peer is online. If anybody has more complete logs that would help in narrowing this down

**Comment by Crypt-iQ** [comment]
I've isolated the issue and will work on getting a patch out. Unfortunately that means 0.15.1 users will have to restart to get the channelupdate to send out

**Comment by BhaagBoseDK** [comment]
> I've isolated the issue and will work on getting a patch out. Unfortunately that means 0.15.1 users will have to restart to get the channelupdate to send out

Wondereful. It happened to me again today and I was coming to post the collected logs for this issue but seems like you are already on top. Could you please explain what is the issue?

**Comment by Crypt-iQ** [comment]
This PR which made LND spec-compliant in one area (https://github.com/lightningnetwork/lnd/pull/6664) calls the function `NotifyWhenOnline` which notifies the caller when the peer is online. In this function, if the peer object is in a map and is actually offline, the caller will hang and never notify when the peer is online. This means that you'd receive the FundingLocked message, and never react to it because the code in #6664 was waiting for the peer to come online. The bug wasn't introduced in #6664 but I wish I had spotted it then, it seems like it's been a long-standing bug that may have affected some other areas of the codebase in a similar way. Basically the access pattern in #6664 of constantly calling `NotifyWhenOnline` and frequent reconnects causes this

**Comment by Roasbeef** [comment]
Thanks for the report all! We have a prelimary fix as part of #6892. Thankfully the fix is actually really small. 

If anyone in this thread is able to apply a live patch, then they can apply this:
```diff
diff --git a/server.go b/server.go
index ad34055d5fa..59d98ec9654 100644
--- a/server.go
+++ b/server.go
@@ -3195,7 +3195,13 @@ func (s *server) NotifyWhenOnline(peerKey [33]byte,
 		select {
 		case <-peer.ActiveSignal():
 		case <-peer.QuitSignal():
-			// The peer quit so we'll just return.
+			// The peer quit so we'll add the channel to the slice
+			// and return.
+			s.mu.Lock()
+			s.peerConnectedListeners[pubStr] = append(
+				s.peerConnectedListeners[pubStr], peerChan,
+			)
+			s.mu.Unlock()
 			return
 		}
 
```

**Comment by blckbx** [comment]
So after some successful channel openings, I got my first case of this kind as well. Unfortunately a restart of LND doesn't fix this. Here's my story. 

Channel openning (remotely initiated, CLN peer):
```
2022-09-17 17:15:33.285 [INF] FNDG: Recv'd fundingRequest(amt=0.05 BTC, push=0 mSAT, delay=144, pendingId=Z) from peer(X)
2022-09-17 17:15:33.586 [INF] FNDG: completing pending_id(Z) with ChannelPoint(CP)
2022-09-17 17:15:33.598 [INF] FNDG: sending FundingSigned for pending_id(Z) over ChannelPoint(CP)
2022-09-17 17:15:33.598 [INF] CNCT: Creating new ChannelArbitrator for ChannelPoint(CP)
2022-09-17 17:15:33.601 [INF] CNCT: ChannelArbitrator(CP): starting state=StateDefault, trigger=chainTrigger, triggerHeight=754513
2022-09-17 17:15:33.602 [INF] NTFN: New spend subscription: spend_id=60, outpoint=CP, script=0 , height_hint=754513
2022-09-17 17:15:33.602 [INF] NTFN: Dispatching historical spend rescan for outpoint=CP, script=0 , start=754513, end=754513
2022-09-17 17:15:33.619 [INF] CNCT: Close observer for ChannelPoint(CP) active
2022-09-17 17:15:33.620 [INF] NTFN: Historical spend dispatch finished for request outpoint=CP, script=0  (start=754513 end=754513) with details: <nil>
2022-09-17 17:15:33.730 [DBG] NTFN: Updated spend hint to height=754513 for unconfirmed spend request outpoint=CP, script=0 
2022-09-17 17:16:54.600 [INF] PEER: Peer(X): unable to read message from peer: EOF
2022-09-17 17:16:54.600 [INF] PEER: Peer(X): disconnecting X, reason: read handler closed
2022-09-17 17:16:56.368 [INF] SRVR: Finalizing connection to X, inbound=true
2022-09-17 17:16:56.584 [WRN] RPCS: Gossip syncer for peer=X not found
2022-09-17 17:16:56.597 [INF] PEER: Peer(X): loading ChannelPoint(CP)
2022-09-17 17:16:56.598 [WRN] PEER: Peer(X): Unable to find our forwarding policy for channel CP, using default values
2022-09-17 17:16:56.598 [INF] PEER: Peer(X): Negotiated chan series queries
2022-09-17 18:02:29.224 [INF] FNDG: ChannelPoint(CP) is now active: ChannelID(W)
2022-09-17 18:02:29.351 [WRN] CHBU: Replacing disk backup for ChannelPoint(CP) w/ newer version
2022-09-17 18:02:29.361 [INF] FNDG: Peer(X) is online, sending FundingLocked for ChannelID(W)
```

Manual reconnect:
```
2022-09-18 13:09:33.830 [INF] SRVR: Disconnecting from X
2022-09-18 13:09:33.830 [INF] PEER: Peer(X): disconnecting X, reason: server: DisconnectPeer called
2022-09-18 13:09:33.830 [INF] PEER: Peer(X): unable to read message from peer: read tcp Y->X: use of closed network connection
2022-09-18 13:09:34.031 [INF] HSWC: ChannelLink(CP): stopping
2022-09-18 13:09:34.032 [INF] HSWC: ChannelLink(CP): exited
2022-09-18 13:09:40.119 [INF] SRVR: Finalizing connection to X, inbound=true
2022-09-18 13:09:40.123 [WRN] RPCS: Gossip syncer for peer=X not found
2022-09-18 13:09:40.348 [INF] PEER: Peer(X): loading ChannelPoint(CP)
2022-09-18 13:09:40.349 [WRN] PEER: Peer(X): Unable to find our forwarding policy for channel CP, using default values
2022-09-18 13:09:40.349 [INF] HSWC: ChannelLink(CP): starting
2022-09-18 13:09:40.350 [INF] HSWC: ChannelLink(CP): HTLC manager started, bandwidth=0 mSAT
2022-09-18 13:09:40.350 [INF] HSWC: ChannelLink(CP): attempting to re-synchronize
2022-09-18 13:09:40.350 [INF] CNCT: Attempting to update ContractSignals for ChannelPoint(CP)
2022-09-18 13:09:40.350 [INF] PEER: Peer(X): Negotiated chan series queries
2022-09-18 13:09:40.452 [INF] HSWC: ChannelLink(CP): received re-establishment message from remote side
2022-09-18 13:09:40.453 [DBG] HSWC: ChannelLink(CP): loaded 0 fwd pks
2022-09-18 13:09:51.711 [ERR] NANN: Unable to retrieve chan status for Channel(CP): edge not found
2022-09-18 13:10:51.804 [ERR] NANN: Unable to retrieve chan status for Channel(CP): edge not found
2022-09-18 13:11:51.790 [ERR] NANN: Unable to retrieve chan status for Channel(CP): edge not found
2022-09-18 13:12:51.713 [ERR] NANN: Unable to retrieve chan status for Channel(CP): edge not found
2022-09-18 13:13:51.796 [ERR] NANN: Unable to retrieve chan status for Channel(CP): edge not found
2022-09-18 13:14:40.352 [ERR] PEER: Peer(X): Unable to enable channel CP: edge not found
```

Restart: 
```
2022-09-18 10:45:48.453 [ERR] NANN: Unable to retrieve chan status for Channel(CP): edge not found
2022-09-18 10:45:58.383 [DBG] HSWC: 
Stopping LND Network Daemon...
2022-09-18 10:45:58.733 [INF] LTND: Received shutdown request.
2022-09-18 10:45:58.733 [INF] LTND: Shutting down...
lnd.service: Deactivated successfully.
Stopped LND Network Daemon.
...
Starting LND Network Daemon...
Attempting automatic RPC configuration to bitcoind
Automatically obtained bitcoind's RPC credentials
2022-09-18 10:46:39.751 [INF] LTND: Version: 0.15.1-beta commit=v0.15.1-beta,...
2022-09-18 10:46:39.751 [INF] LTND: Active chain: Bitcoin (network=mainnet)
...

[INF] CNCT: Close observer for ChannelPoint(CP) active
[INF] CNCT: ChannelArbitrator(CP): starting state=StateDefault, trigger=chainTrigger, triggerHeight=754634
[WRN] NANN: Unable to find channel policies for CP, skipping. This is typical if the channel is in the process of closing.
[WRN] CHBU: Replacing disk backup for ChannelPoint(CP) w/ newer version
[INF] PEER: Peer(X): loading ChannelPoint(CP)
[WRN] PEER: Peer(X): Unable to find our forwarding policy for channel CP, using default values
[ERR] NANN: Unable to retrieve chan status for Channel(CP): edge not found
```

Another restart:
```
2022-09-18 10:45:58.735 [INF] HSWC: ChannelLink(CP): stopping
2022-09-18 10:51:05.968 [DBG] NTFN: Using height hint 754634 retrieved from cache for outpoint=CP, script=0  instead of 754516 for spend subscription
2022-09-18 10:51:05.982 [INF] NTFN: New spend subscription: spend_id=33, outpoint=CP, script=0 , height_hint=754634
2022-09-18 10:51:05.982 [INF] NTFN: Dispatching historical spend rescan for outpoint=CP, script=0 , start=754634, end=754634
2022-09-18 10:51:06.048 [DBG] NTFN: Updated spend hint to height=754634 for unconfirmed spend request outpoint=CP, script=0 
2022-09-18 10:51:06.049 [INF] CNCT: Close observer for ChannelPoint(CP) active
2022-09-18 10:51:06.075 [INF] CNCT: ChannelArbitrator(CP): starting state=StateDefault, trigger=chainTrigger, triggerHeight=754634
2022-09-18 10:51:31.924 [WRN] NANN: Unable to find channel policies for CP, skipping. This is typical if the channel is in the process of closing.
2022-09-18 10:51:32.181 [WRN] CHBU: Replacing disk backup for ChannelPoint(CP) w/ newer version
2022-09-18 10:51:32.197 [INF] SRVR: Finalizing connection to X, inbound=true
2022-09-18 10:51:32.199 [INF] PEER: Peer(X): disconnecting X, reason: server: disconnecting peer X
2022-09-18 10:51:32.203 [INF] SRVR: Finalizing connection to X, inbound=true
2022-09-18 10:51:32.286 [INF] PEER: Peer(X): loading ChannelPoint(CP)
2022-09-18 10:51:32.300 [WRN] PEER: Peer(X): Unable to find our forwarding policy for channel CP, using default values
```

Unsure if this is the same cause as described by the author of this issue but the outcome is almost the same. If unrelated, I can open a separate issue and if required I can provide all data from the logs.
